src/feature/control/control_getinfo.c

```diff
@@ -325,6 +325,33 @@ getinfo_helper_current_time(control_connection_t *control_conn,
   return 0;
 }
 
+/**
+ * Helps to access directory information for a given
+ * type of consensus.
+ *
+ * returns: int
+ *   -1 in cases of an error
+ *   0 otherwise
+ */
+STATIC int
+getinfo_helper_current_consensus(const char* consensus_type,
+                                 char** answer,
+                                 const char** errmsg)
+{
+  tor_mmap_t *mapped = networkstatus_map_cached_consensus(consensus_type);
+  if (mapped) {
+    *answer = tor_memdup_nulterm(mapped->data, mapped->size);
+    tor_munmap_file(mapped);
+    tor_free(mapped);
```

Comment:
Double-free, this line is not in the original code at:
https://github.com/torproject/tor/pull/1328/files#diff-ac2399586b806edb39db242c0a5664bbL589